src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java

```diff
@@ -31,41 +31,50 @@
  * "&lt;" and "&gt;" are correct to the <i>typical</i> convention.
  * The convention is not configurable.
  * </p>
- * <br>
  * <p>
  * Left angle bracket ("&lt;"):
  * </p>
- * <br>
  * <ul>
  * <li> should be preceded with whitespace only
  *   in generic methods definitions.</li>
  * <li> should not be preceded with whitespace
  *   when it is precede method name or following type name.</li>
  * <li> should not be followed with whitespace in all cases.</li>
  * </ul>
- * <br>
  * <p>
  * Right angle bracket ("&gt;"):
  * </p>
- * <br>
  * <ul>
  * <li> should not be preceded with whitespace in all cases.</li>
  * <li> should be followed with whitespace in almost all cases,
  *   except diamond operators and when preceding method name.</li></ul>
- * <br>
  * <p>
  * Examples with correct spacing:
  * </p>
- * <br>
  * <pre>
- * public void &lt;K, V extends Number&gt; boolean foo(K, V) {}  // Generic methods definitions
- * class name&lt;T1, T2, ..., Tn&gt; {}                          // Generic type definition
- * OrderedPair&lt;String, Box&lt;Integer&gt;&gt; p;              // Generic type reference
- * boolean same = Util.&lt;Integer, String&gt;compare(p1, p2);   // Generic preceded method name
- * Pair&lt;Integer, String&gt; p1 = new Pair&lt;&gt;(1, "apple");// Diamond operator
- * List&lt;T&gt; list = ImmutableList.Builder&lt;T&gt;::new;     // Method reference
- * sort(list, Comparable::&lt;String&gt;compareTo);              // Method reference
+ * // Generic methods definitions
```

Comment:
It is not possible to align vertically both
```
<foo> // comment (for HTML)
```
and
```
&lt;foo&gt; // comment (for Java)
```
so all comments are moved to the top..

Reply:
Should we be trying to align the documentation of the java file or should we be trying to align the HTML version of the documention generated from the java file?
I am fine with comments on the top too.

Reply:
@romani What is your opinion?

Reply:
we should prefer to do all possible to make html looks better. As it is the most read place of our documentation. Good looking html and not very good xdoc and javadoc is fine.
Un-aligned javadoc is ok.

In this partucular case it is ok to move comments above code, as we use trailing comments as place for "violation/ok" comments. `// Generic methods definitions` is explanation of code, not a violation marker placeholder.

One day in future, we will udpate this Check documentation to show valid and invalid cases, so trailing placeholders will be required for different purpose. So I am ok with such change.

Reply:
Thanks for the confirmation @romani .